.github/workflows/validate-release.yml

```diff
@@ -0,0 +1,17 @@
+name: Validate release
+
+on:
+  push:
+    branches:
+      - 'releases/**'
```

Comment:
question(non-blocking): This is targeting `main`. Should we also run this check against that branch?

Reply:
One of the two versions being compared comes from the version being embedded in the branch name as in `releases/vX.Y.Z`, so we'd need to compare something else if running on `main`.

Reply:
Echoing that this kind of logic feels useful for releasing, but less useful at any other time. `main` may not point to the latest version if we change something that isn't versioned (like adding a line to the README, or fixing a crash in the example app), and dev/topic branches should be scoped below things versioning

Reply:
Yup was just thinking that we could gate merges to main to make sure we don't unintentionally merge an older version in. Probably overkill though, and def more complicated/would require pinning down what is allowed in `main` based on e.g. commit history and existing release branches, per both of your callouts.